File: apps/android/app/src/main/java/inc/anky/android/core/protocol/AnkyHasher.kt
package inc.anky.android.core.protocol

import java.security.MessageDigest

object AnkyHasher {
    fun sha256Hex(bytes: ByteArray): String {
        val digest = MessageDigest.getInstance("SHA-256").digest(bytes)
        return digest.joinToString(separator = "") { "%02x".format(it) }
    }

    fun sha256Hex(text: String): String = sha256Hex(text.toByteArray(Charsets.UTF_8))
}

File: apps/ios/Anky/Core/Protocol/AnkyHasher.swift
import CryptoKit
import Foundation

public enum AnkyHasher {
    public static func sha256Hex(_ text: String) -> String {
        sha256Hex(Data(text.utf8))
    }

    public static func sha256Hex(_ data: Data) -> String {
        SHA256.hash(data: data)
            .map { String(format: "%02x", $0) }
            .joined()
    }
}
